test/unit/math/mix/fun/copysign_test.cpp

```diff
@@ -1,24 +1,53 @@
 #include <stan/math/mix.hpp>
 #include <gtest/gtest.h>
+#include <cmath>
+#include <complex>
 #include <limits>
 #include <vector>
 
+template <typename U, typename V>
+void expect_eq_signbit(const U& u, const V& v) {
+  using stan::math::signbit;
+  using std::signbit;
+  EXPECT_EQ(signbit(u), signbit(v));
+}
+
 template <typename T>
 void expect_copysign() {
   using stan::math::copysign;
-  using stan::math::signbit;
   using std::copysign;
   using std::numeric_limits;
-  using std::signbit;
-  T inf = numeric_limits<double>::infinity();
-  T nan = std::numeric_limits<double>::quiet_NaN();
-  std::vector<T> ys{inf, -inf, -1, 0, 1};  // inf, -inf, -1.0, 0.0, 1.0 };
 
-  for (const T& x : ys) {
-    for (const T& y : ys) {
-      EXPECT_EQ(signbit(y), signbit(copysign(T(x), y)));
-      EXPECT_EQ(signbit(y), signbit(copysign(x, T(y))));
-      EXPECT_EQ(signbit(y), signbit(copysign(T(x), T(y))));
+  double inf = numeric_limits<double>::infinity();
+  std::vector<double> ys{inf, -inf, -1, 0, 1};
```

Comment:
NaN, -NaN, -0 should go in here too I think

Reply:
Fixed.